src/dvclive/utils.py

```diff
@@ -194,3 +199,39 @@ def read_history(live, metric):
 def read_latest(live, metric_name):
     _, latest = parse_metrics(live)
     return latest["step"], latest[metric_name]
+
+
+def convert_datapoints_to_list_of_dicts(
+    datapoints: List[Dict] | pd.DataFrame | np.ndarray,
+) -> List[Dict]:
+    """
+    Convert the given datapoints to a list of dictionaries.
+
+    Args:
+        datapoints: The input datapoints to be converted.
+
+    Returns:
+        A list of dictionaries representing the datapoints.
+
+    Raises:
+        ValueError: If the `datapoints` argument is not of type pd.DataFrame, np.ndarray, or List[Dict].
+    """
+    if isinstance(datapoints, list):
+        return datapoints
+
+    import pandas as pd
+
+    if isinstance(datapoints, pd.DataFrame):
+        return datapoints.to_dict(orient="records")
```

Comment:
I think you could use this:

https://github.com/iterative/dvclive/blob/19b71b4c05c0e24f747ee4df4e331f960a4dbd97/src/dvclive/utils.py#L153-L157

That way you can check instance types without importing the packages. Hopefully that means we would only import pandas when passed a pandas dataframe.

Reply:
I'll suggest importing at the top and suppressing on import error.

```python
# top level

try:
   import pandas as pd
except ImportError:
   pd = None
```

And guard/check for None.
```python
if pd and isinstance(datapoints, pd.DataFrame):
    pass
```

Reply:
@mnrozhkov I'll defer to @skshetry here as the engineer.

@skshetry would you advise we drop `is_instance_without_import()` (in a separate PR)?

Reply:
@skshetry, thanks for the suggestions! I dropped `is_instance_without_import()` and made it work with this imports: 

```python
if TYPE_CHECKING:
    import numpy as np
    import pandas as pd
else:
    try:
        import pandas as pd
    except ImportError:
        pd = None

    try:
        import numpy as np
    except ImportError:
        np = None
```

It doesn't look nice, but it's the only solution works to meet 2 goals 
- use proper type annotations for `datapoints` 
- meet `mypy` checks

Reply:
It's okay to silence mypy here here.